lib/contentmanager.js

```diff
@@ -881,7 +880,13 @@ ContentManager.prototype.setupRoutes = function () {
 
       that.retrieve(type, query, options, function (error, results) {
         if (error) {
-          res.statusCode = error instanceof ContentTypeError ? 400 : 500;
+          if(error instanceof ContentTypeError) {
+            res.status(400);
+          } else if(error instanceof ContentPermissionError) {
+            res.status(403);
+          } else {
+            res.status(500);
+          }
```

Comment:
If you detest nested `if`s as much as the rest, would be good to reverse the logic here.
```js
if (!error) {
  // @TODO - we need to filter resources according to permissions!
  res.setHeader('Cache-Control', 'no-cache, no-store, must-revalidate');
  ...
}

if(error instance of ContentTypeError) {
  res.status(400);
} else if(error instanceof ContentPermissionError) {
...
```

Reply:
I prefer to keep error handling at the top, hence the nesting